archivebox/index/schema.py

```diff
@@ -51,17 +52,39 @@ def typecheck(self) -> None:
             assert self.output
 
     @classmethod
-    def from_json(cls, json_info):
+    def guess_ts(_cls, dict_info):
+        from ..util import parse_date
+        parsed_timestamp = parse_date(dict_info["timestamp"])
+        start_ts = parsed_timestamp
+        end_ts = parsed_timestamp + timedelta(seconds=int(dict_info["duration"]))
+        return start_ts, end_ts
+
+    @classmethod
+    def from_json(cls, json_info, guess=False):
         from ..util import parse_date
 
         info = {
             key: val
             for key, val in json_info.items()
             if key in cls.field_names()
         }
-        info['start_ts'] = parse_date(info['start_ts'])
-        info['end_ts'] = parse_date(info['end_ts'])
-        info['cmd_version'] = info.get('cmd_version')
+        if guess:
+            keys = info.keys()
+            if "start_ts" not in keys:
+                info["start_ts"], info["end_ts"] = cls.guess_ts(json_info)
+            else:
+                info['start_ts'] = parse_date(info['start_ts'])
+                info['end_ts'] = parse_date(info['end_ts'])
+            if "pwd" not in keys:
+                info["pwd"] = str(os.getcwd() / Path(f"archive/{json_info['timestamp']}"))
```

Comment:
OUTPUT_PATH + ARCHIVE_DIR_NAME instead of cwd + `archive/`

Reply:
Good catch